tests/optim/test_oss.py

```diff
@@ -474,7 +477,11 @@ def closure():
                     dist.gather(p, receptacle, dst=0, group=process_group)
                     if rank == 0:
                         for sync_p in receptacle[1:]:
-                            assert torch.all(torch.eq(receptacle[0], sync_p)), "Models differ in between ranks"
+                            assert torch.all(
```

Comment:
This is non-deterministic failure seems hard to catch. Perhaps you can run this test in a loop for 1000 times just to make sure it is not hitting locally? Otherwise, looks great.

Reply:
no so it was deterministic, it's just that one needed to have the right environment (without cuda), which I didn't have, so to begin with I couldn't repro. Else it's an old test, always been reliable, I've just changed this line so that it displays the values when something is wrong (to check whether it's a tolerance thing or if the values are not related altogether)

Reply:
I see. I was surprised that it failed with py37 not 38.

Reply:
and wasn't failing before.

Reply:
interesting, similar but different place assert:

https://app.circleci.com/pipelines/github/facebookresearch/fairscale/1409/workflows/68a4e48f-1e55-4783-af06-107f70c3690b/jobs/6326